### PR TITLE
Abstract out the different types of node in a zipped Rune

### PR DESCRIPTION
**Pull request by Michael-F-Bryan**
This gives us nice things like the ability to make TensorFlow Lite support optional and helps simplify the `ZuneEngine` code.

### CODE REVIEW

**Review by saidinesh5 — CHANGES_REQUESTED** [review]
This PR isn't compiling. So can't merge this right now. plus Most of these changes will be deleted after the wit file cleanup. So let's not merge this today before the demos and create more work for us.

**Comment by Michael-F-Bryan** [comment]
@saidinesh5 this isn't just a bunch of random refactorings. I'm disabling TensorFlow Lite so we unblock the people who can't work on Weld Studio due to `hotg-runecoral` build issues.

**Pull request merged**
